plugins/imjournal/imjournal.c

```diff
@@ -251,10 +260,26 @@ int sharedJsonProperties)
 		msgAddJSON(pMsg, (uchar*)"!", json, 0, sharedJsonProperties);
 	}
 
+	discardCount = ratelimiter->missed;
 	CHKiRet(ratelimitAddMsg(ratelimiter, NULL, pMsg));
-	STATSCOUNTER_INC(statsCounter.ctrSubmit, statsCounter.mutCtrSubmit);
 
 finalize_it:
+/*
+ * Since ratelimitAddMsg return value does not provide RS_RET_DISCARDMSG
+ * for all cases, we rely on ratelimiter->missed and ratelimiter->done
```

Comment:
can you elaborate on this? Is that a bug that needs fixing?

Reply:
I initially implemented this part based on whether return value is RS_RET_DISCARDMSG and was not seeing discard count change even when it was expected to change. That's when I noticed that ratelimitAddMsg() does not return this value in all cases. 

However, I noticed that this is not true any more. I checked the latest and see it is fixed already:
https://github.com/rsyslog/rsyslog/blob/master/runtime/ratelimit.c#L270

Problem is with my local version that is a few months old. 
I'll update this code. Thanks for catching it.